src/main/java/com/denizenscript/depenizen/bukkit/commands/noteblockapi/NBSCommand.java

```diff
@@ -62,72 +64,59 @@ public NBSCommand() {
     // Use to stop the current song playing for all online players.
     // - nbs stop targets:<server.online_players>
     //
+    // @Usage
+    // Use to play a song only for the linked player in a queue at a specified location.
+    // - nbs play file:MySong at:<[my_location]>
+    //
+    // @Usage
+    // Use to play a song to everyone online if they are 30 blocks away from a specified location.
+    // - nbs play file:MySong targets:<server.online_players> at:<[my_location]> distance:30
     // -->
 
-    private enum Action {PLAY, STOP}
+    public enum Action {PLAY, STOP}
 
-    @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (!scriptEntry.hasObject("targets")
-                    && arg.matchesPrefix("targets", "targets")
-                    && arg.matchesArgumentList(PlayerTag.class)) {
-                scriptEntry.addObject("targets", arg.asType(ListTag.class).filter(PlayerTag.class, scriptEntry));
-            }
-            else if (!scriptEntry.hasObject("file")
-                    && arg.matchesPrefix("file")) {
-                scriptEntry.addObject("file", arg.asElement());
-            }
-            else if (!scriptEntry.hasObject("action")
-                    && arg.matchesEnum(Action.class)) {
-                scriptEntry.addObject("action", arg.asElement());
-            }
-            else {
-                arg.reportUnhandled();
-            }
-        }
-        if (!scriptEntry.hasObject("action")) {
-            throw new InvalidArgumentsException("Action not specified! (play/stop)");
-        }
-        else if (!scriptEntry.hasObject("targets")) {
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("action") Action action,
+                                   @ArgName("file") @ArgPrefixed @ArgDefaultNull String file,
+                                   @ArgName("targets") @ArgPrefixed @ArgDefaultNull @ArgSubType(PlayerTag.class) List<PlayerTag> targets,
+                                   @ArgName("at") @ArgPrefixed @ArgDefaultNull LocationTag location,
+                                   @ArgName("distance") @ArgPrefixed @ArgDefaultText("16") int distance) {
+        if (targets == null) {
             if (Utilities.entryHasPlayer(scriptEntry)) {
-                scriptEntry.addObject("targets", Collections.singletonList(Utilities.getEntryPlayer(scriptEntry)));
+                targets = List.of(Utilities.getEntryPlayer(scriptEntry));
             }
             else {
-                throw new InvalidArgumentsException("Must specify players to add, remove or spectate!");
+                throw new InvalidArgumentsRuntimeException("Must specify players to add or remove!");
```

Comment:
This message still isn't really clear, probably `players to play/stop a song for` or something?